websites/N/Nebula/presence.ts

```diff
@@ -0,0 +1,285 @@
+			break;
+		case "terms":
+			presenceData.details = "Viewing Terms of Service";
+			break;
+		case "privacy":
+			presenceData.details = "Viewing Privacy Policy";
+			break;
+		case "beta":
+			presenceData.details = "Viewing beta apps page";
+			break;
+		case "library":
+			if (path.length === 1) presenceData.details = "Viewing library page";
+			else presenceData.details = getLibraryCategory(path[1].toLowerCase());
+			break;
+		case "explore":
+			if (path.length === 1) presenceData.details = "Viewing explore page";
+			else presenceData.details = getExploreCategory(path[1].toLowerCase());
+			break;
+		case "videos":
+			getVideoDetails(presenceData, showButtons);
+			break;
+		case "search":
+			getSearchDetails(presenceData);
+			break;
+		default:
+			getOtherDetails(presenceData, showButtons);
+			break;
+	}
+}
+
+function getExploreCategory(category: string): string {
+	switch (category) {
+		case "videos":
+			return "Exploring videos";
+		case "channels":
+			return "Exploring channels";
+		case "podcasts":
+			return "Exploring podcasts";
+		case "episodes":
+			return "Exploring episodes";
+	}
+}
+
+function getLibraryCategory(category: string): string {
+	switch (category) {
+		case "latest-videos":
+			return "Viewing latest videos";
+		case "followed-channels":
+			return "Viewing followed channels";
+		case "watch-later":
+			return "Viewing Watch Later";
+		case "watch-history":
+			return "Viewing watch history";
+		case "latest-episodes":
+			return "Viewing latest episodes";
+		case "followed-shows":
+			return "Viewing followed shows";
+		case "saved-episodes":
+			return "Viewing saved episodes";
+		case "listen-history":
+			return "Viewing listening history";
+		case "classes-in-progress":
+			return "Viewing classes in progress";
+		case "saved-classes":
+			return "Viewing saved classes";
+		case "lesson-history":
+			return "Viewing lesson history";
+	}
+}
+
+function getVideoDetails(
+	presenceData: PresenceData,
+	showButtons: boolean
+): void {
+	const videoElement = document.querySelector("video");
+
+	presenceData.details = document.querySelector(
+		"[aria-label='video description'] > div:nth-of-type(1) > h1"
+	).textContent;
+	presenceData.state = document.querySelector(
+		"[aria-label='video description'] > div:nth-of-type(2) > a:nth-of-type(2) > h2"
+	).textContent;
+
+	if (showButtons) {
+		presenceData.buttons = [
+			{
+				label: "Watch Video",
+				url: document.location.href,
+			},
+			{
+				label: "View Channel",
+				url:
+					getRootUrl() +
+					document
+						.querySelector(
+							"[aria-label='video description'] > div > a:nth-of-type(2)"
+						)
+						.getAttribute("href"),
+			},
+		];
+	}
+
+	if (videoElement === null) return;
+	setTimestamps(videoElement, presenceData);
+}
+
+function getSearchDetails(presenceData: PresenceData): void {
+	presenceData.details = "Searching for:";
+	presenceData.state = parseQueryParams().q || "...";
```

Comment:
Optional, but would look ok
```suggestion
	presenceData.state = parseQueryParams().q || "...";
	presenceData.smallImageKey = Assets.Search;
```